.travis.yml

```diff
@@ -128,13 +128,30 @@ jobs:
       language: python
       python: "3.6"
       script:
+        - sed -i "s/.0$/.$TRAVIS_BUILD_NUMBER/" game-core/src/main/resources/META-INF/triplea/product.properties
+        - version=$(cat game-core/src/main/resources/META-INF/triplea/product.properties | sed 's/.* //')
```

Comment:
I'm confused about the sed command at the end, I suppose it's supposed to remove whitespaces from the string if present?
I believe that in case this string ever end up ending with whitespace like `1.2.3    `,  the whole string will be replaced with an empty string because `.` matches any char

Reply:
The regex removes everything up until the last space. The format of the version number has been stable for a relatively long time, ie: `version = 2.0`

Indeed, trailing space would break the regex. I'd be more concerned about the disappearance of the leading space, but again the version number has been quite stable. Given we are unlikely to change version number except for a few times every year, is it worth a more complicated regex?

Reply:
Well we don't really expect the version string itself to contain spaces, so doing a global sed (sed with global flag) to replace spaces with an empty string will do, right? 
Nothing too complicated

Reply:
If there is a trailing space, the previous `sed` higher up in the file is going to fail too:

```
        - ./.travis/setup_gpg
        ## Update product version to include build number.
        ## EG: replace "2.0.0" to be "2.0.15555";
        - sed -i "s/.0$/.$TRAVIS_BUILD_NUMBER/" game-core/src/main/resources/META-INF/triplea/product.properties
```

Reply:
Updated to guard against unexpected white space: 12f07ab83